source/utilities.cc

```diff
@@ -1724,11 +1724,127 @@ namespace aspect
+            maximum_component_value[c] = std::max(maximum_component_value[c], raw_data[dim+c](
+                                                    compute_table_indices(table_points, (components+dim)*idx)));
+        }
+
+      // In case the data is specified on a grid that is equidistant
+      // in each coordinate direction, we only need to store
+      // (besides the data) the number of intervals in each direction and
+      // the begin- and endpoints of the coordinates.
+      // In case the grid is not equidistant, we need to keep
+      // all the coordinates in each direction, which is more costly.
+      // Here we fill the data structures needed for both cases,
+      // and check whether the coordinates are equidistant or not.
+      // We also check the requirement that the coordinates are
+      // strictly ascending.
+
+      // The number of intervals in each direction
+      std::array<unsigned int,dim> table_intervals;
+
+      // Whether or not the grid is equidistant
+      coordinate_values_are_equidistant = true;
+
+      for (unsigned int i = 0; i < dim; i++)
+        {
+          table_intervals[i] = table_points[i] - 1;
+
+          TableIndices<dim> idx;
+          double temp_coord = raw_data[i](idx);
+          double new_temp_coord = 0;
+
+          // The minimum coordinates
+          grid_extent[i].first = temp_coord;
+
+          // The first coordinate value
+          coordinate_values[i].clear();
+          coordinate_values[i].push_back(temp_coord);
+
+          // The grid spacing
+          double grid_spacing = numbers::signaling_nan<double>();
+
+          // Loop over the rest of the coordinate points
+          for (unsigned int n = 1; n < table_points[i]; n++)
+            {
+              idx[i] = n;
+              new_temp_coord = raw_data[i](idx);
+              AssertThrow(new_temp_coord > temp_coord,
+                          ExcMessage ("Coordinates in dimension "
+                                      + int_to_string(i)
+                                      + " are not strictly ascending. "));
+
+              // Test whether grid is equidistant
+              if (n == 1)
+                grid_spacing = new_temp_coord - temp_coord;
+              else
+                {
+                  const double current_grid_spacing = new_temp_coord - temp_coord;
+                  // Compare current grid spacing with first grid spacing,
+                  // taking into account roundoff of the read-in coordinates
+                  if (std::abs(current_grid_spacing - grid_spacing) > 0.005*(current_grid_spacing+grid_spacing))
+                    coordinate_values_are_equidistant = false;
+                }
+
+              // Set the coordinate value
+              coordinate_values[i].push_back(new_temp_coord);
+
+              temp_coord = new_temp_coord;
+            }
+
+          // The maximum coordinate
+          grid_extent[i].second = temp_coord;
+        }
```

Comment:
The code you have up there doesn't actually check whether the data is tensor product. It really only walks down the first row, column, pillar, and never checks any of the other (hopefully redundant) coordinate values. I think that's ok, but it illustrates that the input data format is not well chosen: It allows for redundancies, which you never check, and make implicit assumptions about.

Reply:
agreed. Again, just moved from load_file.